process/block/shardblock.go

```diff
@@ -915,10 +913,14 @@ func (sp *shardProcessor) updateState(headers []data.HeaderHandler, currentHeade
 			log.Debug("could not get shard header from storage")
 			return
 		}
+		currentHeader := headers[i]
```

Comment:
may change for i := range headers to for _, hdr := range headers then this line is not more needed.